pymc3/stats.py

```diff
@@ -478,7 +481,7 @@ def gradient(w):
                      bounds=bounds,
                      constraints=constraints)
 
-        weights = w_fuller(w['x'])
+        weights = w_fuller(np.round(w['x'], round_to))
```

Comment:
Do you have an example for that? Maybe we can solve that my playing with the optimization itself. For example we could try to move the bounds to constraints, with `{'type': 'ineq', 'fun': lambda x: x}` and use an `eq` constraint for the sum instead of two ineq? Then we could also experiment with `slsqp` vs `cobyla`, right?
But to be honest I don't have that much experience with optimization problems, so if you think that the rounding approach is the best way forward, I'm going to believe you :-)